test/macros.jl

```diff
@@ -119,17 +119,20 @@ function macros_test(ModelType::Type{<:JuMP.AbstractModel}, VariableRefType::Typ
 
         cref = @constraint(m, -1 <= x <= 1)
         c = JuMP.constraint_object(cref)
-        @test c.func == x
+        @test c.func isa JuMP.GenericAffExpr # see #1586
```

Comment:
Same comment about the issue links. It doesn't seem necessary to tag these lines. The point is not very subtle and should be explained sufficiently in the comment on `build_constraint`. If I can caricature this practice, imagine 
```
@test 1 + 1 == 2  # see #1234, #1356, #10657
```